crates/crates_io_trustpub/src/access_token.rs

```diff
@@ -0,0 +1,149 @@
+use rand::distr::{Alphanumeric, SampleString};
+use secrecy::{ExposeSecret, SecretString};
+use sha2::digest::Output;
+use sha2::{Digest, Sha256};
+
+/// A temporary access token used to publish crates to crates.io using
+/// the "Trusted Publishing" feature.
+#[derive(Debug)]
+pub struct AccessToken(SecretString);
+
+impl AccessToken {
+    pub const PREFIX: &str = "cio_tp_";
+    const RAW_LENGTH: usize = 31;
+
+    /// Generate a new access token.
+    pub fn generate() -> Self {
+        let raw = Alphanumeric.sample_string(&mut rand::rng(), Self::RAW_LENGTH);
```

Comment:
Just noting for cross-checking purposes: `log2(62) * 31` is approximately 185 bytes of entropy, so more than sufficient 🙂

Reply:
yeah, we're using roughly the same pattern for the regular API keys, though without the checksum, so I figured it should be sufficient :)